File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/websocket/ChatRoomController.java
package com.jmsoftware.maf.springcloudstarter.websocket;

import com.jmsoftware.maf.common.websocket.WebSocketMessagePayload;
import lombok.RequiredArgsConstructor;
import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;
import org.springframework.messaging.handler.annotation.Headers;
import org.springframework.messaging.handler.annotation.MessageMapping;
import org.springframework.messaging.handler.annotation.Payload;
import org.springframework.stereotype.Controller;

import java.util.Map;

/**
 * Description: ChatRoomController, change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 7/15/2021 5:24 PM
 **/
@Slf4j
@Controller
@RequiredArgsConstructor
public class ChatRoomController {
    private final ChatRoomService chatRoomService;

    @SneakyThrows
    @MessageMapping("/send-message")
    public void sendMessage(@Payload WebSocketMessagePayload payload, @Headers Map<String, Object> headers) {
        this.chatRoomService.sendMessage(payload, headers);
    }
}

File: spring-cloud-starter/src/main/kotlin/com/jmsoftware/maf/springcloudstarter/websocket/ChatRoomController.kt
package com.jmsoftware.maf.springcloudstarter.websocket

import com.jmsoftware.maf.common.websocket.WebSocketMessagePayload
import org.springframework.messaging.handler.annotation.Headers
import org.springframework.messaging.handler.annotation.MessageMapping
import org.springframework.messaging.handler.annotation.Payload
import org.springframework.stereotype.Controller

/**
 * # ChatRoomController
 *
 * Description: ChatRoomController, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 10:02 AM
 */
@Controller
class ChatRoomController(
    private val chatRoomService: ChatRoomService
) {
    @MessageMapping("/send-message")
    fun sendMessage(@Payload payload: WebSocketMessagePayload, @Headers headers: Map<String, Any>) {
        chatRoomService.sendMessage(payload, headers)
    }
}
